server/src/main/java/org/elasticsearch/action/admin/cluster/remote/RemoteClusterNodesAction.java

```diff
@@ -0,0 +1,132 @@
+import org.elasticsearch.action.support.HandledTransportAction;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.transport.BoundTransportAddress;
+import org.elasticsearch.common.util.concurrent.ThreadContext;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.transport.RemoteClusterPortSettings;
+import org.elasticsearch.transport.TransportInfo;
+import org.elasticsearch.transport.TransportService;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.elasticsearch.transport.RemoteClusterPortSettings.REMOTE_CLUSTER_SERVER_ENABLED;
+
+public class RemoteClusterNodesAction extends ActionType<RemoteClusterNodesAction.Response> {
+
+    public static final RemoteClusterNodesAction INSTANCE = new RemoteClusterNodesAction();
+    public static final String NAME = "cluster:admin/remote_cluster/nodes";
+
+    public RemoteClusterNodesAction() {
+        super(NAME, RemoteClusterNodesAction.Response::new);
+    }
+
+    public static class Request extends ActionRequest {
+
+        public static final Request INSTANCE = new Request();
+
+        public Request() {}
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            return null;
+        }
+    }
+
+    public static class Response extends ActionResponse {
+
+        private final List<DiscoveryNode> nodes;
+
+        public Response(List<DiscoveryNode> nodes) {
+            this.nodes = nodes;
+        }
+
+        public Response(StreamInput in) throws IOException {
+            super(in);
+            this.nodes = in.readList(DiscoveryNode::new);
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeList(nodes);
+        }
+
+        public List<DiscoveryNode> getNodes() {
+            return nodes;
+        }
+    }
+
+    public static class TransportAction extends HandledTransportAction<Request, Response> {
+
+        private final TransportService transportService;
+
+        @Inject
+        public TransportAction(TransportService transportService, ActionFilters actionFilters) {
+            super(RemoteClusterNodesAction.NAME, transportService, actionFilters, Request::new);
+            this.transportService = transportService;
+        }
+
+        @Override
+        protected void doExecute(Task task, Request request, ActionListener<Response> listener) {
+            final NodesInfoRequest nodesInfoRequest = new NodesInfoRequest();
+            nodesInfoRequest.clear();
+            nodesInfoRequest.addMetrics(NodesInfoRequest.Metric.SETTINGS.metricName(), NodesInfoRequest.Metric.TRANSPORT.metricName());
```

Comment:
For now the remote cluster information is derived from both settings and transport info. But we should augument NodesInfo with a dedicated RemoteCluster section which can then be used directly here.